src/shipit_code_coverage_backend/shipit_code_coverage_backend/api.py

```diff
@@ -75,8 +75,10 @@ def coverage_latest():
 
 def phabricator_base_revision_from_phid(revision_phid):
     try:
-        revision = phabricator.get_base_revision(secrets.PHABRICATOR_TOKEN, revision_phid)
-        if revision:
+        phabricator = PhabricatorAPI(secrets.PHABRICATOR_TOKEN)
+        diff = phabricator.load_diff(revision_phid=revision_phid)
+        revision = diff.get('baseRevision')
+        if revision and revision_exists_on_central(revision):
```

Comment:
I'd move the revision exists on central check to phabricator.py, as it's needed for both code coverage and static analysis (if the revision doesn't exist on mozilla-central, then we can't use it)

Reply:
I prefer to check on SA with the local repo, and i find the Api client code better without too many external integrations (like hgweb)

Reply:
We can check with a local repo if available and with hgweb if the local repo is not available